### ISSUE TITLE
SetUserCustomStatus results in user_not_found error

### ISSUE THREAD

**Issue by hackedd** [slack-go/slack#809]
Recently `SetUserCustomStatus` started failing with a `user_not_found` error.

It looks like the `users.profile.set` no longer accepts an empty string for the `user` parameter. When I remove this parameter from the API call, it seems to work.

As a workaround, I'm now calling `SetUserCustomStatusContext` with my own User ID (which works even though the API docs say "this argument may only be specified by team admins")

**Comment by sryoya** [comment]
I could find the request with  an empty `user` got `user_not_found` even with `curl`

Without user param
```
$ curl -X POST "https://slack.com/api/users.profile.set?token=xxx&profile=%7B%22status_text%22%3A%20%22test%22%2C%20%22status_emoji%22%3A%20%22%22%2C%20%22status_expiration%22%3A%201%7D&user&22&22&pretty=1"

-> receives user_not_found
```

With an empty user param
```
$ curl -X POST "https://slack.com/api/users.profile.set?token=xxx&profile=%7B%22status_text%22%3A%20%22test%22%2C%20%22status_emoji%22%3A%20%22%22%2C%20%22status_expiration%22%3A%201%7D&pretty=1"

-> success
```

Regardless of the above result, as it's an optional field ([Slack API doc](https://api.slack.com/methods/users.profile.set)), I think we should omit if the value is empty.

**Comment by sryoya** [comment]
Created the PR #868 to fix this along with #867
